src/WeightVectors.jl

```diff
@@ -212,15 +215,65 @@ Base.iszero(w::AbstractWeightVector) = w.m[2] == 5
     sample_within_level(rng, m, pos, len)
 end
 
-@inline function sample_within_level(rng, m, pos, len)
+@inline function sample_within_level(rng, m::Memory{UInt64}, pos::UInt64, len::UInt64)
+    shift = leading_zeros(len-1)
+
+    # These should never happen but is required to make `@inbounds` safe:
+    1 <= pos::UInt64 <= UInt64(2)^63 || @fail 0
```

Comment:
```suggestion
    1 <= pos <= UInt64(2)^63 || @fail 0
```

the `pos` type is already constrained in the signature

Reply:
This is here so that point (4) in the proof can reference a single piece of source code to establish type and value. I agree that it isn't strictly necessary and that it is a no-op.

Reply:
ok, perfect, let's just let the compiler to do its work :-)